iml-sfa/src/main.rs

```diff
@@ -36,7 +36,7 @@ enum ImlSfaError {
 
 fn create_retry_endpoint_policy<E: Debug>(len: u32) -> impl RetryPolicy<E> + Send {
     move |k: u32, e| {
-        if k < len {
+        if k < len - 1 {
```

Comment:
What if `len==0`? What would it mean?

Reply:
When len is 0 that would be the first attempt at making the call.

Reply:
I misread this earlier. Basically the policy is used to limit how many times we retry hitting an endpoint. So if we create a policy with a length of two then that means it will try to hit the endpoint twice. If we create a policy with a length of zero then it would never attempt to hit the endpoint. I'll add a unit test to cover this.

Reply:
I would expect that retry mean literally retry, and thus `len` would mean the allowed number of failures: 0 mean hit just one time, 1 - retry one one if failed, and so on.